web3_proxy/src/bin/web3_proxy_cli/change_user_admin_status.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::Context;
+use argh::FromArgs;
+use entities::{admin, login, user};
+use ethers::types::Address;
+use log::{debug, info};
+use migration::sea_orm::{
+    self, ActiveModelTrait, ColumnTrait, DatabaseConnection, EntityTrait, ModelTrait, IntoActiveModel,
+    QueryFilter,
+};
+use web3_proxy::frontend::errors::FrontendErrorResponse;
+
+/// change a user's admin status. eiter they are an admin, or they aren't
+#[derive(FromArgs, PartialEq, Eq, Debug)]
+#[argh(subcommand, name = "change_admin_status")]
+pub struct ChangeUserAdminStatusSubCommand {
+    /// the address of the user whose admin status you want to modify
+    #[argh(positional)]
+    address: String,
+
+    /// true if the user should be an admin, false otherwise
+    #[argh(positional)]
+    should_be_admin: bool,
+}
+
+impl ChangeUserAdminStatusSubCommand {
+    pub async fn main(self, db_conn: &DatabaseConnection) -> anyhow::Result<()> {
+        let address: Address = self.address.parse()?;
+        let should_be_admin: bool = self.should_be_admin;
+
+        let address: Vec<u8> = address.to_fixed_bytes().into();
+
+        // Find user in database
+        let user = user::Entity::find()
+            .filter(user::Column::Address.eq(address.clone()))
+            .one(db_conn)
+            .await?
+            .context("No user found with that address")?;
+
+        // Check if there is a record in the database
+        let mut admin = admin::Entity::find()
+            .filter(admin::Column::UserId.eq(address))
+            .all(db_conn)
+            .await?;
+
+        debug!("user: {:#?}", user);
+
+        match admin.pop() {
+            None if should_be_admin => {
+                // User is not an admin yet, but should be
+                let new_admin = admin::ActiveModel {
+                    user_id: sea_orm::Set(user.id),
+                    ..Default::default()
+                };
+                new_admin.insert(db_conn).await?;
+            },
+            Some(old_admin) if !should_be_admin => {
+                // User is already an admin, but shouldn't be
+                old_admin.delete(db_conn).await?;
+            },
+            _ => {}
+        }
+
+        // Remove any user logins from the database (incl. bearer tokens)
+        let delete_result = login::Entity::delete_many()
+            .filter(login::Column::UserId.eq(user.id))
+            .exec(db_conn)
+            .await;
+
+        debug!("cleared modified logins: {:?}", delete_result);
```

Comment:
also clear redis.

since we do need this same code here, in a frontend endpoint, and similar logic in the /logout endpoint (that just clears the one bearer token though), should we make a helper function?

Reply:
i don't think you need to do anything with the salt. the salted stuff is just to keep track that the user has logged in. we use that for our own tracking. we don't want to delete anything from there

Reply:
i thought we had the bearer tokens cached in redis. that's what i was referring to. i might have deleted that when we moved everything to the db, but I don't think so.

Reply:
yh looks like it was deleted in that case. I will remove this segment too then.